frontend/src/lib/stores.js

```diff
@@ -1,12 +1,31 @@
-import words from '../languages/et/words.json'
 import { writable } from 'svelte/store'
+import languages from '../languages/languages.json'
+
+let wordCache
+let cachedLanguage
+
+async function getWords() {
+  if (cachedLanguage !== _currentLanguage.code) {
+    wordCache = (await import(`../languages/${_currentLanguage.code}/words.json`))
+      .default
+    cachedLanguage = _currentLanguage.code
+  }
+  return wordCache
+}
+
+export const currentLanguage = writable(languages[0])
+let _currentLanguage
+currentLanguage.subscribe(async (value) => {
+  _currentLanguage = value
+  await getWords()
+})
 
 function randInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min
 }
 
 function getRandomWord() {
-  return words[randInt(0, words.length - 1)]
+  return wordCache[randInt(0, wordCache.length - 1)]
```

Comment:
I am probably missing something. The app breaks at this line because `wordCache` is `undefined`.

Reply:
I think we have to initialize `wordCache`

Reply:
I didn't say you should directly access `wordCache`, but `const words = await getWords()` and `getWords` has the caching logic inside it

Reply:
where am I going to call this `const words = await getWords()`? in the `getRandomWord `function?_ it has to be in an async function.

Reply:
yes

Reply:
okay resolved.